### DISCUSSION TITLE
Error running on Raspberry Pi

### DISCUSSION THREAD

**Discussion by sepsky** [danielpaulus/go-ios#261] [Q&A]
Hi @danielpaulus Happy Friday :) 

I'm trying to run go-ios on my raspberry pi, so that I can connect my iOS device to it and run automation from it. I've installed go-ios by cloning the repo, and building the executable. 

My device is connected to RaspPi, and I can access the content on the device. 

I tried running `ios list` to get the devices, but I got this message:
```
{"err":"dial unix /var/run/usbmuxd: connect: no such file or directory","level":"fatal","msg":"failed getting device list","time":"2023-05-26T16:23:38+01:00"}
```
I got the same error when I tried to activate a device using the UDID: 
```
ios activate --udid=XYZ
```

is this because I need to install usbmuxd? I'm sorry this sounds a bit elementary, but if I can add any more context please let me know :) 

PS. This is the verbose log I get: 
```
{"level":"info","msg":"Set Debug mode","time":"2023-05-26T16:53:40+01:00"}
{"level":"debug","msg":"map[--all:false --apps:false --arg:[] --basedir:\u003cnil\u003e --binary:false --bundle-id:\u003cnil\u003e --bundleid:\u003cnil\u003e --certfile:\u003cnil\u003e --details:false --dstPath:\u003cnil\u003e --env:[] --force:false --gpxfilepath:\u003cnil\u003e --help:false --lang:false --lat:\u003cnil\u003e --list:false --locale:false --lon:\u003cnil\u003e --nojson:false --orgname:\u003cnil\u003e --output:\u003cnil\u003e --p12file:\u003cnil\u003e --password:\u003cnil\u003e --path:\u003cnil\u003e --pid:\u003cnil\u003e --plist:false --port:\u003cnil\u003e --pretty:false --process:\u003cnil\u003e --r:false --setlang:\u003cnil\u003e --setlocale:\u003cnil\u003e --skip:[] --skip-all:false --srcPath:\u003cnil\u003e --stop-at-entry:false --stream:false --system:false --test-runner-bundle-id:\u003cnil\u003e --testrunnerbundleid:\u003cnil\u003e --trace:false --udid:\u003cnil\u003e --verbose:true --version:false --wait:false --xctest-config:\u003cnil\u003e --xctestconfig:\u003cnil\u003e \u003capp_path\u003e:\u003cnil\u003e \u003cbundleID\u003e:\u003cnil\u003e \u003ccwd\u003e:\u003cnil\u003e \u003chost\u003e:\u003cnil\u003e \u003chostPort\u003e:\u003cnil\u003e \u003ckey\u003e:[] \u003cpass\u003e:\u003cnil\u003e \u003cpattern\u003e:\u003cnil\u003e \u003cport\u003e:\u003cnil\u003e \u003cprofileFile\u003e:\u003cnil\u003e \u003cprofileId\u003e:\u003cnil\u003e \u003cprofileName\u003e:\u003cnil\u003e \u003cprofileTypeId\u003e:\u003cnil\u003e \u003csrcpattern\u003e:\u003cnil\u003e \u003ctarget\u003e:\u003cnil\u003e \u003ctargetPort\u003e:\u003cnil\u003e \u003cuser\u003e:\u003cnil\u003e activate:false add:false apps:false assistivetouch:false auto:false ax:false batterycheck:false cp:false crash:false create-cert:false date:false debug:false devicename:false devicestate:false diagnostics:false disable:false diskspace:false dproxy:false enable:false erase:false forward:false fsync:false get:false httpproxy:false image:false info:false install:false instruments:false ip:false kill:false lang:false launch:false list:true listen:false ls:false mkdir:false mobilegestalt:false mount:false notifications:false pair:false pcap:false prepare:false printskip:false profile:false ps:false pull:false push:false readpair:false reboot:false remove:false resetlocation:false rm:false runtest:false runwda:false screenshot:false setlocation:false setlocationgpx:false syslog:false toggle:false tree:false uninstall:false version:false voiceover:false zoomtouch:false]","time":"2023-05-26T16:53:40+01:00"}
{"err":"dial unix /var/run/usbmuxd: connect: no such file or directory","level":"fatal","msg":"failed getting device list","time":"2023-05-26T16:53:40+01:00"}
```
Thanks a lot and have a great day

**Comment by wellkilo** [comment]
Yes. go-ios uses the usbmuxd socket on Linux, and this error means the daemon/socket is missing before go-ios can even talk to the phone:

    dial unix /var/run/usbmuxd: connect: no such file or directory

Install and start usbmuxd plus libimobiledevice support using your Raspberry Pi distribution packages, then verify in this order:

    sudo systemctl enable --now usbmuxd
    systemctl status usbmuxd
    ls -l /var/run/usbmuxd /run/usbmuxd
    idevice_id -l
    ios list

Also install the Apple USB udev rules supplied by the distro packages, unlock the iPhone, and accept the trust prompt for normal pairing. Avoid running two usbmuxd instances against the same device.

On most current Linux systems /var/run points to /run. If the service is active but only /run/usbmuxd exists, check that compatibility symlink and whether go-ios/usbmuxd are running in different containers/namespaces.

After ios list works, iOS 17+ operations that use RemoteXPC require the tunnel daemon too:

    sudo ios tunnel start

That requirement is in the current [README](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/README.md#L9-L13). Fix the missing usbmuxd transport first; higher-level activation/screenshot debugging only makes sense after device discovery succeeds.
